lib/infrastructure/pigeons/scan_provider.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:cobble/domain/entities/pebble_scan_device.dart';
+import 'package:cobble/infrastructure/pigeons/pigeons.dart';
+import 'package:hooks_riverpod/all.dart';
+
+class ScanState {
+  final bool scanning;
+  final List<PebbleScanDevice> devices;
+
+  ScanState(this.scanning, this.devices);
+}
+
+class _ScanCallbacks extends StateNotifier<ScanState> implements ScanCallbacks {
+  _ScanCallbacks()
+      : super(
+          ScanState(
+            true,
```

Comment:
Shouldn't default state be not scanning?

Reply:
Yup, it's probably better because we don't scan by default.